src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java

```diff
@@ -0,0 +1,109 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.bigquery.stepsdesign;
+
+import com.google.cloud.bigquery.FieldValue;
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.TableResult;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import io.cdap.e2e.utils.BigQueryClient;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * BQValidation.
+ */
+public class BigQueryMultiTableValidation {
+  static Gson gson = new Gson();
+  public static boolean validateBQToBigQueryMultiTable(List<String> sourceTables, List<String> targetTables)
+    throws IOException, InterruptedException {
+    // Check if there's only one source table
+    if (sourceTables.size() == 1) {
+      String currentTargetTable = targetTables.get(0);
+      return validateBQToBQSingleTable(sourceTables.get(0), currentTargetTable);
+    } else {
+      if (sourceTables.size() != targetTables.size()) {
+        throw new IllegalArgumentException("Number of source tables and target tables must be the same.");
+      }
+      // Iterate over each pair of source and target tables
+      for (int i = 0; i < sourceTables.size(); i++) {
+        String currentSourceTable = sourceTables.get(i);
+        String currentTargetTable = targetTables.get(i);
+        // Perform validation for the current pair of tables
+        if (!validateBQToBQSingleTable(currentSourceTable, currentTargetTable)) {
+          return false; // Return false if validation fails for any table pair
+        }
+      }
+      // Return true if validation passes for all table pairs
+      return true;
+    }
+  }
+
+  public static boolean validateBQToBQSingleTable(String sourceTable, String targetTable)
+    throws IOException, InterruptedException {
+    // Fetch data from the source BigQuery table
+    List<Object> bigQueryRows = new ArrayList<>();
+    getBigQueryTableData(sourceTable, bigQueryRows);
+
+    // Convert fetched data into JSON objects
+    List<JsonObject> bigQueryResponse = new ArrayList<>();
+    for (Object row : bigQueryRows) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(row), JsonObject.class);
+      bigQueryResponse.add(jsonData);
+    }
+    // Fetch data from the target BigQuery table
+    List<Object> bigQueryRows2 = new ArrayList<>();
+    getBigQueryTableData(targetTable, bigQueryRows2);
+    List<JsonObject> bigQueryResponse2 = new ArrayList<>();
+    for (Object row : bigQueryRows2) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(row), JsonObject.class);
```

Comment:
nit: You can create a proper class here to deserialize.

Reply:
we are using these method from common code that we are using for all the plugins to deserialize data from bq.